### PR TITLE
LESS: add new DS6 classes, mixins and variables for typography #39

### PR DESCRIPTION
**Pull request by ianmcburnie**
See issue #39 for spec.

Still awaiting clarification on what font-weight `secondary` is, and whether Market Sans even supports font weight other than regular or bold.

**Question**: are we okay with the classnames `medium`, `regular`, `small` etc. Or are they too generic? I'm kind of okay with them them, but maybe `medium-text`, `regular-text`, `small-text` is another alternative.

screen shot 2018-02-05 at 1 44 40 pm

screen shot 2018-02-05 at 1 45 03 pm

### CODE REVIEW

**Comment by seangates** [comment]
Usually the steps are: x-small, small, medium, large, x-large. "Regular" sort of seems out of place in that.

**Comment by seangates** [comment]
I know you've matched the typography PDF, but I wonder about those step names ...

**Comment by ianmcburnie** [comment]
Yeah, `regular` seems weird to me as well. Name wise it seems too close to `medium`. But there has already been a LONG design system thread on the names, and this is what they want, so in the interest of moving this forward would like to just get this in.

**Comment by seangates** [comment]
That is ... unfortunate. Onward!

File: src/less/typography/ds6/typography-base.less [hunk 1]
```diff
@@ -1,47 +1,27 @@
...
-    .ds6-typography-caption-2;
-}
-
-.caption-3 {
-    .ds6-typography-caption-3;
-}
-
-.legal-1 {
-    .ds6-typography-legal-1;
+.small {
```
Comment (seangates) [inline src/less/typography/ds6/typography-base.less]: Yes, let's go with the `-text` postfix. E.g.: `.large-text-1`, `.small-text`, etc.

In your doc examples, the class doesn't really describe what we're making large or small, which might be confusing when used.

Comment (ianmcburnie) [reply src/less/typography/ds6/typography-base.less]: I'm on it.

**Pull request merged**